Keas.Mvc/ClientApp/components/Keys/KeySerialTable.tsx

```diff
@@ -16,142 +20,184 @@ interface IProps {
   onEdit?: (keySerial: IKeySerial) => void;
 }
 
-interface IFilter {
-  id: string;
-  value: any;
+interface IFilterOption {
+  value: string;
+  displayText: string;
 }
 
-interface IRow {
-  original: IKeySerial;
+// UI for Key status column filter
+const KeyStatusColumnFilter = ({
+  column: { filterValue, setFilter }
+}) => {
+  // Render a multi-select box
+  return (
+    <select
+      value={filterValue}
+      style={{ width: '100%' }}
+      onChange={e => {
+        setFilter(e.target.value || undefined);
+      }}
+    >
+      {ReactTableStatusUtil.defaultFilterOptions.map(option => (
+        <option key={option.value} value={option.value}>
+          {option.displayText}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+// Logic to control what rows get displayed
+const statusFilter = (rows: any[], id, filterValue) => {
+  if (filterValue === 'all') {
+    return rows;
+  }
+  if (filterValue === 'active') {
+    return rows.filter(r => getRowStatus(r) === 'Active');
+  }
+  if (filterValue === 'inactive') {
+    return rows.filter(
+      r => getRowStatus(r) === 'Lost' || getRowStatus(r) === 'Destroyed'
+    );
+  }
+  if (filterValue === 'special') {
+    return rows.filter(r => getRowStatus(r) === 'Special');
+  }
+  return rows;
+};
+
+const getRowStatus = (row: any) => row.original?.status;
+
+class ReactTableStatusUtil {
```

Comment:
I think you can dump everything in here.  same with the other places you did the same, like serials and such